procrastinate/aiopg_connector.py

```diff
@@ -39,6 +40,43 @@ async def wrapped(*args, **kwargs):
     return wrapped
 
 
+def wrap_query_exceptions(coro: CoroutineFunction) -> CoroutineFunction:
+    """
+    Detect ConnectorException's caused by psycopg2 OperationalError with a "server
+    closed the connection unexpectedly" message and retry a number of times.
+
+    This decorator is for decorating functions already decorated with wrap_exceptions.
+    """
+
+    @functools.wraps(coro)
+    async def wrapped(*args, **kwargs):
+        final_exc = None
+        cnt = 0
+        max_tries = (
+            args[0]._pool.maxsize + 1 if args and getattr(args[0], "_pool", None) else 1
+        )
+        while cnt < max_tries:
+            try:
+                return await coro(*args, **kwargs)
+            except exceptions.ConnectorException as exc:
+                exc_cause = exc.__cause__
```

Comment:
put the decorators the other way around, it will be easier :)

Reply:
Yeah, I had figured this out :-)